### DISCUSSION TITLE
Finding Overridable Members From a ITypeSymbol

### DISCUSSION THREAD

**Discussion by JasonBock** [dotnet/roslyn#47631] [General] [answered]
I'm currently working on a source generator where I need to get a list of all methods from an `ITypeSymbol` reference that could/should be overridden in a new type that derives from that given type (eventually I'll need to get all the properties and events as well). As I'm finding out, this isn't trivial :). I noticed that VS has a feature that creates all the overrides needed for either an interface or a class, and I'm wondering if that exists in Roslyn proper such that it's not tied specifically to VS.

I'm basically looking to not re-invent the wheel if possible, so if there's a way to do this programmatically, please let me know, TIA.

**Comment by PathogenDavid** [comment, marked as answer]
The abstract class implementation codefix appears to be in [Features/Core/Portable/ImplementAbstractClass](https://github.com/dotnet/roslyn/tree/6ce8dedddd3811ac0ba975838d011d09a94e3a03/src/Features/Core/Portable/ImplementAbstractClass) with the C#-specific stuff in [Features/CSharp/Portable/ImplementAbstractClass/CSharpImplementAbstractClassCodeFixProvider.cs](https://github.com/dotnet/roslyn/blob/63bea1d1d104234c8b945c7a7d3afec5001b950f/src/Features/CSharp/Portable/ImplementAbstractClass/CSharpImplementAbstractClassCodeFixProvider.cs)

And the interface implementation codefix is in [src/Features/Core/Portable/ImplementInterface](https://github.com/dotnet/roslyn/tree/63bea1d1d104234c8b945c7a7d3afec5001b950f/src/Features/Core/Portable/ImplementInterface) with some C#-specific stuff in [src/Features/CSharp/Portable/ImplementInterface](https://github.com/dotnet/roslyn/tree/63bea1d1d104234c8b945c7a7d3afec5001b950f/src/Features/CSharp/Portable/ImplementInterface)

**Reply by JasonBock** [reply]
Thanks, I'll take a look at that.
